src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt

```diff
@@ -78,6 +79,8 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
                 reporter.logFailedIteration(minimizedFailedIteration)
                 return minimizedFailedIteration
             }
+            // Reset range after each generated scenario to start from initial range bounds on next scenario and avoid very diverse values
```

Comment:
// Reset the parameter generator ranges to start with the same initial bounds on each scenario generation.